### PR TITLE
populate: support multiple public, cluster networks in engulf (bsc#1076010)

### PR DESCRIPTION
**Pull request by tserong**
This removes the assumption in the engulfer that there can be no more
than one public or cluster network.  Without this change, running engulf
on a cluster with (e.g.) public_network = 10.27.21.0/24, 10.27.20.0/24
will result in the following error:

  [ERROR   ] Failed to determine cluster's public_network.
  [ERROR   ] Failed to replace derived public_network with public_network
             of existing cluster.

Note that this still leaves the public_network and cluster_network settings
as strings in the pillar data (changing the pillar data to be an array would
have been somewhat more invasive).

Fixes: https://bugzilla.suse.com/show_bug.cgi?id=1076010
Fixes: https://github.com/SUSE/DeepSea/issues/582
Signed-off-by: Tim Serong <tserong@suse.com>

### CODE REVIEW

File: srv/modules/runners/populate.py [hunk 1]
```diff
@@ -1009,30 +1006,31 @@ def _get_existing_cluster_network(addrs, public_network=None):
...
-        # We have multiple possible networks.  This is liable to happen with OSDs
-        # when there is a private cluster network.  Let's try to remove the public
-        # network.
-        minion_networks = [n for n in minion_networks if n != public_network]
-        network = minion_networks[0] if len(set(minion_networks)) == 1 else None
+    # Collapse minion_networks back to unique items.  Usually there'll
+    # be only one network after this, unless the cluster is using a
+    # separate public and private network, or is using multiple public
+    # and/or private networks
+    minion_networks = list(set(minion_networks))
```
Comment (jan--f) [inline srv/modules/runners/populate.py]: Any reason you're not using ```minion_networks = set()``` right away?

Comment (tserong) [reply srv/modules/runners/populate.py]: Just that when I wrote it I was thinking "minion_networks" needs to always be a list. I now realise that the `[n for n in minion_networks]` business on the next line would work just fine against a set, and would return a list, so `list()` in the above line is redundant.

**Comment by tserong** [comment]
I took the liberty of adding a 'TODO' comment to investigate simplifying `_get_existing_cluster_networks`. I've also gotten rid of that redundant `list()`

**Comment by tserong** [comment]
please run basic suite

**Comment by tserong** [comment]
`2018-02-21 23:00:13,534.534 INFO:teuthology.openstack:2018-02-22 04:00:13,527.527 ERROR:teuthology.lock:Could not contact lock server: http://localhost:8080/`

That can't possibly be a result of this commit, can it?

**Comment by jan--f** [comment]
@tserong that wouldn't be my first guess either. @smithfarm what do you think?

**Comment by smithfarm** [comment]
My guess is teuthology is hosed and will need to be redeployed.

**Comment by kshtsk** [comment]
please run health-ok suite

**Comment by kshtsk** [comment]
please run openattic suite

**Comment by kshtsk** [comment]
please run rgw suite

**Comment by kshtsk** [comment]
please run health-ok suite

**Comment by kshtsk** [comment]
please run openattic suite

**Comment by tserong** [comment]
Anyone up for reviewing this?

**Comment by tserong** [comment]
Rebased

**Pull request merged**